hibernate-core/src/main/java/org/hibernate/SharedSessionContract.java

```diff
@@ -158,4 +158,16 @@
 	 */
 	@Deprecated
 	Criteria createCriteria(String entityName, String alias);
+
+	/**
+	 * Set the preferred JDBC batch size for the current Session.
+	 * Overrides the global JDBC batch size defined by the {@code hibernate.default_batch_fetch_size} for the scope of the current {@code Session}.
+	 *
+	 * @param batchSize Session-level JDBC batch size
+	 *
+	 * @since 5.2
+	 *
+	 * @see org.hibernate.cfg.AvailableSettings#DEFAULT_BATCH_FETCH_SIZE
+	 */
+	void setBatchSize(int batchSize);
```

Comment:
1. For consistency, I'd prefer `#setJdbcBatchSize`
2. Wouldn't it also make sense to expose a getJdbcBatchSize()?
3. I think the more correct JavaDoc link would be to `org.hibernate.boot.spi.SessionFactoryOptions#getJdbcBatchSize` and/or `org.hibernate.boot.SessionFactoryBuilder#applyJdbcBatchSize`